core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -0,0 +1,179 @@
+          ))
+          _ = logger.info(s"Creating workspace record [workspaceId = ${workspaceId}]")
+          savedWorkspace <- traceDBIOWithParent("saveNewWorkspaceToRawlsDB", parentSpan)(_ => {
+            createMultiCloudWorkspaceInDatabase(
+              workspaceId.toString,
+              workspaceRequest,
+              dataAccess,
+              parentSpan)
+          })
+          response <- traceDBIOWithParent("doOp", parentSpan)(_ => op(savedWorkspace))
+        } yield {
+          response
+        }
+      }
+    }
+  }
+
+  private def getCloudContextCreationStatus(workspaceId: UUID,
+                                            jobControlId: String,
+                                            accessToken: OAuth2BearerToken): Future[CreateCloudContextResult] = {
+    val result = workspaceManagerDAO.getWorkspaceCreateCloudContextResult(
+      workspaceId, jobControlId, accessToken
+    )
+    result.getJobReport.getStatus match {
+      case StatusEnum.SUCCEEDED => Future.successful(result)
+      case _ => Future.failed(new WorkspaceManagerPollingOperationException(result.getJobReport.getStatus))
+    }
+  }
+
+  private def jobStatusPredicate(t: Throwable): Boolean = {
+    t match {
+      case t: WorkspaceManagerPollingOperationException => (t.status == StatusEnum.RUNNING)
+      case _ => false
+    }
+  }
+
+  private def pollCloudContext(workspaceId: UUID, jobControlId: String, accessToken: OAuth2BearerToken): Future[Unit] = {
+    for {
+      result <- retryUntilSuccessOrTimeout(pred = jobStatusPredicate)(2 seconds, multiCloudWorkspaceConfig.cloudContextPollTimeout) {
+        () => getCloudContextCreationStatus(workspaceId, jobControlId, accessToken)
+      }
+    } yield {
+      result match {
+        case Left(_) => throw new CloudContextCreationFailureException(
+          s"Cloud context creation failed [workspaceId=${workspaceId}, jobControlId=${jobControlId}]",
+          workspaceId,
+          jobControlId
+        )
+        case Right(_) => ()
+      }
+    }
+  }
+
+  private def createMultiCloudWorkspaceInDatabase(workspaceId: String,
+                                                  workspaceRequest: MultiCloudWorkspaceRequest,
+                                                  dataAccess: DataAccess,
+                                                  parentSpan: Span = null): ReadWriteAction[Workspace] = {
+    val currentDate = DateTime.now
+    val workspace = Workspace(
+      namespace = workspaceRequest.namespace,
+      name = workspaceRequest.name,
+      workspaceId = workspaceId,
+      bucketName = "",
+      workflowCollectionName = Some(workspaceId),
+      createdDate = currentDate,
+      lastModified = currentDate,
+      createdBy = userInfo.userEmail.value,
+      attributes = workspaceRequest.attributes,
+      isLocked = false,
+      workspaceVersion = WorkspaceVersions.V2,
+      googleProjectId = GoogleProjectId(""),
```

Comment:
It's weird to see all these Google properties in a MC Workspace. Is it necessary to specify these dummy values, or can default be used (or another constructor added).